### DISCUSSION TITLE
Bash shfmt options

### DISCUSSION THREAD

**Discussion by dwooldridge-ctoken** [zed-industries/zed#19296] [Feature Requests]
Hi, I'm trying to figure out how to configure the `bash-language-server` to set some `shfmt` arguments. I've tried all options I can think of based off the config defined here: 

https://github.com/bash-lsp/bash-language-server/blob/main/server/src/config.ts

But can't get it to work. Formatting continues to use the default settings.

The docs don't have any examples of a list of config options: https://zed.dev/docs/languages/bash

The bash extension docs are also lacking: https://github.com/d1y/bash.zed

Has anyone figured out how to get this working?

I have tried many different settings:

```json
  "lsp": {
    "bash-language-server": {
      "settings": {
        "bash": {
          "shfmt": {
            "languageDialect": "bash",
            "binaryNextLine": true,
            "caseIndent": true
          }
        }
      }
    }
```

```json
  "lsp": {
    "bash-language-server": {
      "shfmt": {
        "languageDialect": "bash",
        "binaryNextLine": true,
        "caseIndent": true
      }
    }
  }

```

**Comment by BeyondEvil** [comment]
Any success here?

**Reply by dwooldridge-ctoken** [reply]
No, haven't managed to figure this out yet

**Comment by BeyondEvil** [comment]
Seems like I solved it @dwooldridge-ctoken 

```json
  "languages": {
    "Shell Script": {
      "format_on_save": "on",
      "formatter": {
        "external": {
          "command": "shfmt",
          "arguments": [
            "-bn",
            "-ci",
            "-sr",
            "-i",
            "2"
          ]
        }
      }
    }
  },
```

Turns out the language isn't "Bash", but "Shell Script".

**Reply by dwooldridge-ctoken** [reply]
Nice, this works for formatting, but I'd like to be able to support doing this with the language server, but after looking at the extension source, it doesn't seem possible at this time and it looks like it's now unmaintained, so will use this method.

**Reply by injust** [reply]
By the way, you should add `["--filename", "{buffer_path}"]` to the arguments. See https://zed.dev/docs/languages/sh#formatting for an example of how to configure shfmt as an external formatter.

The shfmt man page says:

```
--filename str
Provide a name for the standard input file.
Use of this flag is necessary for EditorConfig support to work with stdin, since EditorConfig files are found relative to the location of a script.
```

Also see the [Zed docs](https://zed.dev/docs/reference/all-settings#formatter) about the `{buffer_path}` placeholder.

**Comment by dwooldridge-ctoken** [comment]
Closing due to @BeyondEvil's workaround of calling `shfmt` directly instead of via the bash language server.
